graviti/manager/common.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python3
+#
+# Copyright 2022 Graviti. Licensed under MIT License.
+#
+
+"""Common tools."""
+
+import warnings
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from graviti.manager.commit import Commit
+
+
+class StatusWarning(Warning):
+    """This class defines the warning that the commit of dataset branch or tag is not up-to-date.
+
+    Arguments:
+       revision_type: The type of current dataset revision.
+       name: The name of current dataset revision.
+
+    """
+
+    def __init__(self, revision_type: str, name: str):
+        super().__init__()
+        self._type = revision_type
```

Comment:
we have a enum for revision_type, shall we use it here?

Reply:
we can not import `RevisionType` in `common.py`, this will cause a `circular import`. And we can not move the `RevisionType` to `common.py`. 
RevisionType -> need Branch -> need check_head_status -> need RevisionType